docs/proposal-01-multi-profile-repo.md

```diff
@@ -0,0 +1,273 @@
+cmd/config/v0.9.11
+cmd/config/v0.9.10
+```
+
+The prefix before the semver version corresponds to the directory in which the product lives, with the last directory also being the name of the component. For example the repository contains:
+```
+tree kustomize/
+├── api
+    └── main.go
+    ...
+├── cmd
+│   └── config
+	└── main.go
+	...
+└── kyaml
+    └── main.go
+    ...
+```
+
+If you wanted to get version `v0.8.9` of `api` you would checkout to tag `api/v0.8.9` and inside the `api/` directory it would contain the desired code.
+The case of `cmd/config` is slightly less clean, it's easier to write automation around detecting new profiles if the profile directory is always at the top
+level directory. Support for sub-directories is in the [Open Questions / Known Unknowns](#open-questions--known-unknowns) section below.
+
+This approach could be used for versioning profiles in a repository.
+
+### Profile Subscription modifications
+Currently we only support referencing a repository as follows:
+```
+apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  name: nginx-profile-test
+  namespace: default
+spec:
+  profileURL: https://github.com/weaveworks/nginx-profile
+  branch: main
+```
+
+We want to maintain support for `branch` workflows as it's useful for development, while also introducing a new `tags` approach. There are two possible approaches:
+
+#### Approach 1
+
+```yaml
+apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  name: foo-test
+  namespace: default
+spec:
+  profileURL: https://github.com/weaveworks/weaveworks-profiles
+  version: v0.1.2
+  profile: foo
+```
+
+```yaml
+apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  name: bar-test
+  namespace: default
+spec:
+  profileURL: https://github.com/weaveworks/weaveworks-profiles
+  branch: main
+  profile: bar
+```
+
+This approach hides the fact that the git tag contains the profile name, and leaves it up to the profiles controller to concatenate the `profile` and `tag` value together.
+This introduces a common `profile` (or `profileName`) field that is shared across the two for us to know which profile in the repository we are referencing and its directory (they must be equal).
+
+#### Approach 2
+
+```yaml
+apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  name: foo-test
+  namespace: default
+spec:
+  profileURL: https://github.com/weaveworks/weaveworks-profiles
+  version: foo/v0.1.2
+```
+
+```yaml
+apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  name: bar-test
+  namespace: default
+spec:
+  profileURL: https://github.com/weaveworks/weaveworks-profiles
+  branch: main
+  path: bar/
+```
+
+This approach makes it clear upon inspection what tag is actually being referenced in the repository, but is less clear what directory/profile inside the repository is being referenced.
+It also maintains two completely different approaches for using `branch` vs `tag`
```

Comment:
I prefer this option, path is more explicit than inferring from the tag, tagging strategies vary wildly and we can't really rely on it. Instead of a tag, you could ref a SHA (which is more reliable) and there's no path inferrable.

Reply:
Would you mind explaining what do you mean under tagging strategies vary wildly and that we can't rely on it?
I mean we apply the strategy. We create tags and we create the folder structure for it. And if it's documented, why can't we rely on it? Do you mean that people will have difficulties adjusting to it and using it in their own repos and profiles and that we can't rely on proper adoption?

Reply:
There are many tag naming policies (we are not going to be the only folks maintaining profiles).

So, it's not about _us_ relying on _our_ tag naming strategy, it's other folks... :-)

Reply:
> Do you mean that people will have difficulties adjusting to it and using it in their own repos and profiles and that we can't rely on proper adoption?

Ok, so you're are worried about adoption part. Makes sense. But Kubernetes is widely adopted and being contributed too. We could argue that this scheme is already pretty well known?

Reply:
Adding the option for `SHA` sounds like a nice follow up feature :wink: 

> But Kustomize is widely adopted and being contributed too. We could argue that this scheme is already pretty well known?

Anecdotal but before Kevin pointed me at this repo I'd never seen this pattern before :smile: 

I think with either approach its up to us to heavily document and explain the process to the users, so it doesn't matter too much

Reply:
I am happy with the greater clarity of approach 2 👍